### DISCUSSION TITLE
Formatting in VSCode for templ + tailwindcss?

### DISCUSSION THREAD

**Discussion by benrbowers** [a-h/templ#882] [General]
The recommended way to format .templ files in VSCode is to install the the templ extension and then set the formatter for templ files to "a-h.templ".
It works great for general html, my only gripe with this formatter is that there is no way to integrate with tailwindcss.

Ideally, I would like the "reordering" of tailwind classes that is provided by the `prettier-plugin-tailwindcss` for prettier.

Does anyone know if either:
1. I can get the same functionality with the "a-h.templ" formatter?
2. I can use prettier with .templ files in VSCode?

Thanks!

**Comment by noor-tg** [comment]
did you find anything ?

**Comment by gregmulvaney** [comment]
I would try out eslint with autofix enabled. There is a [Eslint tailwindcss plugin](https://www.npmjs.com/package/eslint-plugin-tailwindcss) and eslint is generally less opinionated than prettier.

**Reply by noor-tg** [reply]
After my last comment . I searched github and found prettier plugin that work with templ.

Thanks @gregmulvaney

**Reply by benrbowers** [reply]
@noor-tg Can I ask what plugin it is?

**Reply by noor-tg** [reply]
https://github.com/llamux/prettier-plugin-tailwindcss-extra

**Comment by sturmenta** [comment]
I think this is related: [vscode: Integrations w/ Prettier, Emmet, etc](https://github.com/a-h/templ/issues/230)
